src/orange/orangeinp/ConvexRegion.cc

```diff
@@ -471,13 +449,42 @@ void GenTrap::build(ConvexSurfaceBuilder& insert_surface) const
         auto j = (i + 1) % lo_.size();
         Real3 const ilo{lo_[i][0], lo_[i][1], -hz_};
         Real3 const jlo{lo_[j][0], lo_[j][1], -hz_};
+        Real3 const jhi{hi_[j][0], hi_[j][1], hz_};
         Real3 const ihi{hi_[i][0], hi_[i][1], hz_};
 
         // Calculate outward normal by taking the cross product of the edges
-        auto normal = make_unit_vector(cross_product(jlo - ilo, ihi - ilo));
+        auto lo_normal = make_unit_vector(cross_product(jlo - ilo, ihi - ilo));
+        auto hi_normal = make_unit_vector(cross_product(ihi - jhi, jlo - jhi));
 
-        // Insert the plane
-        insert_surface(Sense::inside, Plane{normal, ilo});
+        // Insert appropriate service, planar or "twisted"
+        if (soft_equal(dot_product(lo_normal, hi_normal), real_type{1}))
+        {
+            // Insert a real plane
+            insert_surface(Sense::inside, Plane{lo_normal, ilo});
+        }
+        else
+        {
+            // Determine coefficients
+            auto alo = jlo[1] - ilo[1];
+            auto ahi = jhi[1] - ihi[1];
+            auto blo = -(jlo[0] - ilo[0]);
+            auto bhi = -(jhi[0] - ihi[0]);
```

Comment:
```suggestion
            auto blo = ilo[0] - jlo[0];
            auto bhi = ihi[0] - jhi[0];
```
Otherwise you introduce signed zeros.